pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -111,6 +112,19 @@ func GetPodConditionFromList(conditions []v1.PodCondition, conditionType v1.PodC
 	return -1, nil
 }
 
+func (pc *PodCache) labelFilter(old, cur interface{}) bool {
+	oldPod := old.(*v1.Pod)
+	curPod := cur.(*v1.Pod)
+
+	// If labels updated, trigger proxy push
+	if curPod.Status.PodIP != "" && !reflect.DeepEqual(oldPod.Labels, curPod.Labels) {
+		pc.proxyUpdates(curPod.Status.PodIP)
```

Comment:
>Why is this a filter? It seems like it should be added to the queue and processed in onEvent?

This is a good question, i meant to do that. But when implementing, found in the queue, we do not pass the original object, so we can not compare old and new object labels

Reply:
I think we can refactor the queue interface if more such cases occur

Reply:
hmm.. I thought most core k8s controllers are written in this model (just get the name of the resource in the queue). How do they handle these types of things?

IIRC the reason we remove the old object is because it makes ordering dependency which breaks SyncAll?